Add eventStream signature to ResponseFactory contract

csfh opened the pull request:
**Commit description**

Adds `eventStream` to `Illuminate\Contracts\Routing\ResponseFactory` in order to add type support for this method, aligning this with the same behavior that we have for `stream` of which `eventStream` is a convenient wrapper.

**Why this change**

At this time the `ResponseFactory` contract does not include a signature for the `eventStream` method. However, the function does exist on the underlying concrete implementation returned by the `response()` helper.

Not having this included can cause some false positive warnings and errors when statically analyzing a codebase using the `eventStream` function, such as when using intellisense in VS Code. Additionally, this can throw off AI agents from using the method, even if directly instructed to (tested using Claude Sonnet 4).

The method is already described on the `Response` facade–this PR would align the response helper with the facade.

This is a additive feature and does not introduce breaking changes.

**Of note:**

I do not see any existing tests for `eventStream`, but here is additionally PRs for context on the function.

- https://github.com/laravel/framework/pull/54776
- https://github.com/laravel/framework/pull/55141

crynobone commented on src/Illuminate/Contracts/Routing/ResponseFactory.php:
```diff
@@ -57,6 +57,16 @@ public function json($data = [], $status = 200, array $headers = [], $options =
+     * @param  \Illuminate\Http\StreamedEvent|string|null  $endStreamWith
+     * @return \Symfony\Component\HttpFoundation\StreamedResponse
+     */
+    public function eventStream(Closure $callback, array $headers = [], StreamedEvent|string|null $endStreamWith = '</stream>');
```
We never add new method to an interface on patch/minor release. This has to go to `master` branch.

csfh replied:
Thanks! I'll take care of that. It's in draft right now but any comments would be beyond appreciated!

csfh replied:
I've applied the patch onto the latest `master` and changed the base branch on the PR. I also realized I had included the type hints in the method signature. This was removed since the contract file does not reference types directly in the code other than primitives.

csfh committed:
Add eventStream signature to ResponseFactory contract

Adds `eventStream` to `Illuminate\Contracts\Routing\ResponseFactory` in order to add type support for this method, aligning this with the same behavior that we have for `stream` of which `eventStream` is a convenient wrapper.
